src/components/pages/PointSystem/TaskDialog.vue

```diff
@@ -0,0 +1,196 @@
+<template>
+  <dialog-base
+    :visible="visible"
+    @update:visible="handleVisibleChange"
+    :title="pointsForCategory.titleProgress"
+    class="task-dialog"
+  >
+    <div>
+      <p>Complete {{ pointsForCategory.titleProgress }}-related tasks in order to level up your skill</p>
+      <div class="task-dialog__card-progress">
+        <div class="current-level">
+          <p>Your level</p>
+          <p>LVL {{ levelCurrent }} <span>/ LVL 6</span></p>
+        </div>
+        <div class="progress-container">
+          <div class="progress-bar" :style="{ width: progressPercentage + '%' }"></div>
+        </div>
+
+        <div class="next-level">
+          <p>Next level reward</p>
+          <p>{{ pointsForCategory.nextLevelRewardPoints }}</p>
+        </div>
+      </div>
+      <div class="task-dialog__card-current">
+        <div class="img-title">
+          <token-logo v-if="isTokenImage" :token="this.getImageSrc(imageName)" size="small" />
+          <img v-else :src="this.getImageSrc(imageName)" :alt="imageName" />
+          <p>{{ pointsForCategory.titleTask }}</p>
+        </div>
+        <p class="description">{{ pointsForCategory.descriptionTask }}</p>
+        <s-divider />
+        <p class="currently-amount">
+          Currently: <span> ${{ this.pointsForCategory.currentProgress.toFixed(2) }}</span>
+        </p>
+      </div>
+    </div>
+  </dialog-base>
+</template>
+
+<script lang="ts">
+import { mixins, components } from '@soramitsu/soraneo-wallet-web';
+import { Component, Mixins, Prop } from 'vue-property-decorator';
+
+import { getImageSrc, isTokenImage } from '@/consts/pointSystem';
+import { CalculateCategoryPointResult } from '@/types/pointSystem';
+
+@Component({
+  components: {
+    DialogBase: components.DialogBase,
+    TokenLogo: components.TokenLogo,
+  },
+})
+export default class TaskDialog extends Mixins(mixins.TranslationMixin) {
+  @Prop({ required: true }) readonly pointsForCategory!: CalculateCategoryPointResult;
+  @Prop({ required: true }) readonly visible!: boolean;
+  @Prop({ required: true }) readonly onClose!: () => void;
+
+  private handleVisibleChange(newVal: boolean) {
```

Comment:
Any issues here to use `DialogMixin`? onClose  & close event duplicates functionality